aten/src/ATen/native/sparse/SparseCsrTensorMath.cpp

```diff
@@ -85,6 +132,19 @@ bool is_square_or_vec(int64_t dim_i, int64_t dim_j, int64_t dim_k) {
   return (dim_i == dim_k  && dim_k == dim_j) || (dim_i == dim_j && dim_k == 1);
 }
 
+Tensor& sin_sparse_csr_out(const Tensor& self, Tensor& result) {
+  return unary_op_out(&at::sin_outf, self, result);
+}
+
+Tensor sin_sparse_csr(const Tensor& self) {
+  auto result = get_result_tensor_for_unary_op(&at::sin, self);
+  return sin_sparse_csr_out(self, result);
```

Comment:
Shouldn't it be just `return result`? `get_result_tensor_for_unary_op` already gives a sparse tensor with `result.values() == sin(self.values())`

Reply:
Good observation, thanks Ivan. This should be fixed now.